File: src/main/java/me/leodev/trollingfreedomreborn/trolls/movement/ForceJump.java
package me.leodev.trollingfreedomreborn.trolls.movement;

import me.leodev.trollingfreedomreborn.main.Core;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerMoveEvent;

import java.util.ArrayList;

public class ForceJump implements Listener {

    public static ArrayList<String> Jump1 = new ArrayList();

    public void Jump(Player p) {
        Player p2 = p.getPlayer();
        Jump1.add(p2.getName());
    }

    public void UnJump(Player p) {
        Player p2 = p.getPlayer();
        Jump1.remove(p2.getName());
    }

    @EventHandler
    public void onWalk(PlayerMoveEvent e) {
        Player p = e.getPlayer();
        if (Jump1.contains(p.getName())) {
            if (!e.getFrom().toVector().equals(e.getTo().toVector()))
                p.setVelocity(p.getVelocity().setY(1.0));
            int id = Bukkit.getScheduler().scheduleSyncDelayedTask(Core.instance, () -> p.setVelocity(p.getVelocity().setY(-1.0)), 20L);
            Core.instance.addTask(p, "forcejump", id);
        }
    }
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/trolls/movement/ForceJump.kt
package com.leomadrassi.trollingfreedomreborn.trolls.movement

import com.leomadrassi.trollingfreedomreborn.main.Core
import org.bukkit.Bukkit
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerMoveEvent

class ForceJump : Listener {
    companion object {
        val Jump1: MutableList<String> = ArrayList()
    }

    fun Jump(p: Player) {
        Jump1.add(p.name)
    }

    fun UnJump(p: Player) {
        Jump1.remove(p.name)
    }

    @EventHandler
    fun onWalk(e: PlayerMoveEvent) {
        val p = e.player
        if (Jump1.contains(p.name)) {
            if (!e.from.toVector().equals(e.to.toVector()))
                p.setVelocity(p.velocity.setY(1.0))
            val id = Bukkit.getScheduler().scheduleSyncDelayedTask(Core.instance, Runnable { p.setVelocity(p.velocity.setY(-1.0)) }, 20L)
            Core.instance.addTask(p, "forcejump", id)
        }
    }
}
